it/uk/gov/hmrc/incorporatedentityidentificationfrontend/api/controllers/JourneyControllerISpec.scala

```diff
@@ -74,18 +110,55 @@ class JourneyControllerISpec extends ComponentSpecHelper with JourneyStub with I
   }
 
   "POST /api/limited-company-journey" should {
-    "return CREATED and supply a journey start url" in {
-      stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
-      stubCreateJourney(CREATED, Json.obj("journeyId" -> testJourneyId))
+    "return a created journey" when {
+      "all the URLs provided are relative" in {
+        stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+        stubCreateJourney(CREATED, Json.obj("journeyId" -> testJourneyId))
+
+        lazy val result = post("/incorporated-entity-identification/api/limited-company-journey", testJourneyConfigJson)
+
+        (result.json \ "journeyStartUrl").as[String] must include(appRoutes.CaptureCompanyNumberController.show(testJourneyId).url)
+        result.status mustBe CREATED
+
+        await(journeyConfigRepository.findJourneyConfig(testJourneyId, testInternalId)) mustBe Some(testLimitedCompanyJourneyConfig)
+      }
+      "a URL provided in non-relative but in allowed hosts list" in {
+        val testJourneyConfigJson: JsObject = Json.obj(
+          "continueUrl" -> testContinueUrl,
+          "deskProServiceId" -> testDeskProServiceId,
+          "signOutUrl" -> testSignOutUrl,
+          "accessibilityUrl" -> testLocalAccessibilityUrl,
+          "regime" -> testRegime
+        )
+        stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+        stubCreateJourney(CREATED, Json.obj("journeyId" -> testJourneyId))
+
+        lazy val result = post("/incorporated-entity-identification/api/journey", testJourneyConfigJson)
+
+        (result.json \ "journeyStartUrl").as[String] must include(appRoutes.CaptureCompanyNumberController.show(testJourneyId).url)
 
-      lazy val result = post("/incorporated-entity-identification/api/limited-company-journey", testJourneyConfigJson)
+        result.status mustBe CREATED
+      }
+    }
+    "return Bad Request" when {
+      "one URL provided for the journey config is not relative" in {
+        val testJourneyConfigJson: JsObject = Json.obj(
+          "continueUrl" -> testContinueUrl,
+          "deskProServiceId" -> testDeskProServiceId,
+          "signOutUrl" -> testSignOutUrl,
+          "accessibilityUrl" -> testStagingAccessibilityUrl,
```

Comment:
Hi Evi if I use "/" what should I have 400 or 500?
I think 400 .... RedirectUrl(url) that we are using for checking the validity of a url it does not accept a simple "/" ... it will throw an IllegalArgumentExcpetion I think ..... have a look to its code ....
For RedirectUrl(url)  a simple "/" is invalid so I think we should send back 400.

<img width="1002" alt="Screenshot 2022-04-27 at 13 17 57" src="https://user-images.githubusercontent.com/4958044/165516378-640b0c1a-db97-4211-b9e8-f5c16c9982b6.png">

dont know if it is a big deal but If I use "/" in your test I get 500 not 400.

<img width="786" alt="Screenshot 2022-04-27 at 13 19 47" src="https://user-images.githubusercontent.com/4958044/165516682-34b8b885-4ecb-47fe-9fc6-c44a833707bb.png">

 have a look and let me know what you think